src/openms/source/ANALYSIS/OPENSWATH/DIAScoring.cpp

```diff
@@ -394,37 +400,13 @@ namespace OpenMS
       // create the theoretical distribution from the sum formula
       EmpiricalFormula empf(sum_formula);
       isotope_dist = empf.getIsotopeDistribution(CoarseIsotopePatternGenerator(dia_nr_isotopes_));
+      iso_.renormalize(isotopes, isotope_dist);
```

Comment:
here, the CoarseIsotopePatternGenerator is still called.
Are you sure this branch is not taken?
In any case, replace it with Cache...